From 1a16a5db70f8fcff58b59cfbe737e9416858db30 Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang.wiwynn@gmail.com>
Date: Wed, 26 Mar 2025 18:49:46 +0800
Subject: [PATCH 203/203] Add retry attempts configuration for fan sensors

Added a Meson option and updated FanMain.cpp to allow configuring the
number of retry attempts for fan sensor operations.
---
 meson.build         | 1 +
 meson.options       | 7 +++++++
 src/fan/FanMain.cpp | 4 ++--
 3 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 1b30ce6..26a5301 100644
--- a/meson.build
+++ b/meson.build
@@ -28,6 +28,7 @@ add_project_arguments(
     '-DBOOST_ALLOW_DEPRECATED_HEADERS',
     '-DBOOST_ASIO_HAS_IO_URING',
     '-DBOOST_ASIO_DISABLE_EPOLL',
+    '-DFAN_SENSOR_RETRY_ATTEMPTS=' + get_option('fan_sensor_retry_attempts').to_string(),
     language: 'cpp',
 )
 
diff --git a/meson.options b/meson.options
index ca4a0c3..dfc7bc6 100644
--- a/meson.options
+++ b/meson.options
@@ -119,3 +119,10 @@ option(
     value: 'false',
     description: 'Disable in2_alarm event.',
 )
+option(
+    'fan_sensor_retry_attempts',
+    type: 'integer',
+    value: 0,
+    description: 'Number of retry attempts for fan sensor operations',
+)
+
diff --git a/src/fan/FanMain.cpp b/src/fan/FanMain.cpp
index b946f56..142d431 100644
--- a/src/fan/FanMain.cpp
+++ b/src/fan/FanMain.cpp
@@ -671,7 +671,7 @@ int main()
 
     boost::asio::post(io, [&]() {
         createSensors(io, objectServer, tachSensors, pwmSensors, presenceGpios,
-                      systemBus, nullptr);
+                      systemBus, nullptr, FAN_SENSOR_RETRY_ATTEMPTS);
     });
 
     boost::asio::steady_timer filterTimer(io);
@@ -693,7 +693,7 @@ int main()
                     return;
                 }
                 createSensors(io, objectServer, tachSensors, pwmSensors,
-                              presenceGpios, systemBus, sensorsChanged, 5);
+                              presenceGpios, systemBus, sensorsChanged, FAN_SENSOR_RETRY_ATTEMPTS);
             });
         };
 
-- 
2.43.0

